protocol/rpcprovider/provider_state_machine.go

```diff
@@ -0,0 +1,83 @@
+package rpcprovider
+
+import (
+	"context"
+	"time"
+
+	"github.com/lavanet/lava/v2/protocol/chainlib"
+	"github.com/lavanet/lava/v2/protocol/chainlib/chainproxy/rpcclient"
+	"github.com/lavanet/lava/v2/protocol/common"
+	"github.com/lavanet/lava/v2/protocol/lavaprotocol"
+	"github.com/lavanet/lava/v2/utils"
+	pairingtypes "github.com/lavanet/lava/v2/x/pairing/types"
+)
+
+type RelaySender interface {
+	SendNodeMsg(ctx context.Context, ch chan interface{}, chainMessage chainlib.ChainMessageForSend, extensions []string) (relayReply *chainlib.RelayReplyWrapper, subscriptionID string, relayReplyServer *rpcclient.ClientSubscription, proxyUrl common.NodeUrl, chainId string, err error)
+}
+
+type ProviderStateMachine struct {
+	relayRetriesManager *lavaprotocol.RelayRetriesManager
+	chainId             string
+}
+
+func NewProviderStateMachine(chainId string) *ProviderStateMachine {
+	return &ProviderStateMachine{
+		relayRetriesManager: lavaprotocol.NewRelayRetriesManager(),
```

Comment:
let it get the relay retries manager as an interface, and get it in the constructor, easier to mock in tests or wrap in the future